README.md

```diff
@@ -272,58 +333,151 @@ tracking revenue that is not actually being generated.
+[event addPartnerParameter:@"foo" value:@"bar"];
+
 [Adjust trackEvent:event];
 ```
 
 You can read more about special partners and these integrations in our [guide to special partners][special-partners].
 
+### <a id="session-parameters">Session parameters
+
+Some parameters are saved to be send in every event and session of the adjust SDK. Once you have add any of these 
+parameters, you don't need to add them every time, since they will be saved locally. If you add the same parameter twice, 
+there will be no effect.
+
+These session parameters can be called before the adjust SDK is launched to make sure they are send even on install. If you
+need to send them with an install, but can only obtain the needed values after launch, it's possible to 
+[delay](#delay-start) the first launch of the adjust SDK to allow this behaviour.
+
+### <a id="session-callback-parameters"> Session callback parameters
+
+The same callback parameters that are registered for [events](#callback-parameters) can be also saved to be sent in every 
+event or session of the adjust SDK.
+
+The session callback parameters have a similar interface of the event callback parameters. Instead of adding the key and 
+it's value to an event, it's added through a call to `Adjust` method `addSessionCallbackParameter:value:`:
+
+```objc
+[Adjust addSessionCallbackParameter:@"foo" value:@"bar"];
+```
+
+The session callback parameters will be merged with the callback parameters added to an event. The callback parameters 
+added to an event have precedence over the session callback parameters. Meaning that, when adding a callback parameter to 
+an event with the same key to one added from the session, the value that prevails is the callback parameter added to the 
+event.
+
+It's possible to remove a specific session callback parameter by passing the desiring key to the method 
+`removeSessionCallbackParameter`.
+
+```objc
+[Adjust removeSessionCallbackParameter:@"foo"];
+```
+
+If you wish to remove all key and values from the session callback parameters, you can reset it with the method 
+`resetSessionCallbackParameters`.
+
+```objc
+[Adjust resetSessionCallbackParameters];
+```
+
+### <a id="session-partner-parameters">Session partner parameters
+
+In the same way that there is [session callback parameters](#session-callback-parameters) that are send every in event or 
+session of the adjust SDK, there is also session partner parameters.
+
+These will be transmitted to network partners, for the integrations that have been activated in your adjust [dashboard].
+
+The session partner parameters have a similar interface of the event partner parameters. Instead of adding the key and it's
+value to an event, it's added through a call to `Adjust` method `addSessionPartnerParameter:value:`:
+
+```objc
+[Adjust addSessionPartnerParameter:@"foo" value:@"bar"];
+```
+
+The session partner parameters will be merged with the partner parameters added to an event. The partner parameters added 
```

Comment:
From here there is a double up, also below the delay is mentioned but differently to above....??